python/lsst/ts/salobj/topics/write_topic.py

```diff
@@ -111,6 +114,17 @@ def has_data(self):
         """Has `data` ever been set?"""
         return self._has_data
 
+    async def close(self):
+        """Shut down and release resources.
+
+        Intended to be called by SalInfo.close(),
+        since that tracks all topics.
+        """
+        if not self.isopen:
+            return
+        self.isopen = False
+        self._writer.close()
```

Comment:
I actually prefer this way because there's no point trying again if the call to `_writer.close()` fails. The flag is intended to mean "I made an effort to close this topic". I suppose another option is:
```
try:
    self._writer.close()
finally:
    self.isopen = False
```

Reply:
Oh, I like that latter version. I'm thinking of a situation where the first attempt to close fails due to something timing out, but that it might be possible to close after that. I don't know if that's possible.

Reply:
It is not possible.

Reply:
Keep in mind that these close methods are almost always called as the system is being shut down, so cleanup will happen in any case. A lot of this is just to be nice and avoid unpleasant warnings in unit tests.